Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:20.04
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install dependencies for building the robot_model library
+RUN apt-get update && apt-get install -y \
+    gcc \
+    g++ \
+    make \
+    git \
+    wget \
+    cmake \
+    && rm -rf /var/lib/apt/lists/*
+
+# import previously downloaded packages
+RUN mkdir -p lib
+WORKDIR lib
+COPY ./source/ .
+# install libraries and dependencies
+RUN . ./install.sh
```

Comment:
I would suggest the Dockerfile actually invokes the sub-installations individually over multiple RUN commands (state rep, dynamical systems, then robot model). I would say the high-level install script is there as a convenience to the user who wants to install all (or in future selectively install with some args), but the Dockerfile can be a bit smarter / more verbose than that. 
Benefits: you can benefit from caching, it's much easier to see where any build stages fail, and it leaves the docker file more configurable if you want to break it into build stages or change install behaviour modules by ARGS
Drawbacks: minor code duplication of the contents / logic of the install script

Reply:
makes sense I will do that